apps/web/src/blocks/HeroBlock/index.tsx

```diff
@@ -128,28 +153,51 @@ function HeroBlock({
   title,
   subTitle,
   cta,
-  input,
+  textinput,
   image,
-  blockConfig
+  blockConfig,
+  contentAlign,
+  layout
 }: HeroBlockProps) {
-  const layout = blockConfig?.layout || 'imgRight';
+  const isFullBleed =
+    typeof image === 'object' ? image?.imageProps?.fill : false;
+
+  const contentPosition = layout || 'contentLeft';
+  const alignText = contentAlign || 'left';
 
-  const className = genClassName([layout]);
   return (
-    <StyledWrapper className={className} gutter={false} fullBleed>
-      <ImageWrapper image={image} $layout={layout} />
-
-      <ContentWrapper className={className}>
-        {eyebrow && <Eyebrow>{eyebrow}</Eyebrow>}
-        {title && <Title>{title}</Title>}
-        {subTitle && <SubTitle {...subTitle} />}
-        {(input || cta) && (
-          <InputWrapper className={className}>
-            {input?.type && <Input {...input} />}
-            {cta && <CtaButton cta={cta} />}
-          </InputWrapper>
-        )}
-      </ContentWrapper>
+    <StyledWrapper
+      gutter={false}
+      {...blockConfig}
+      hidden={blockConfig?.hidden ?? false}
+    >
+      <InnerWrapper
+        $layout={contentPosition}
+        $isFullBleed={isFullBleed || false}
+      >
+        <ContentWrapper
+          $layout={contentPosition}
+          $isFullBleed={isFullBleed || false}
+          $contentAlign={alignText}
+        >
+          {eyebrow && <Eyebrow>{eyebrow}</Eyebrow>}
+          {title && <Title>{title}</Title>}
+          {subTitle && <SubTitle {...subTitle} />}
+          {textinput && cta && (
+            <InputWrapper $contentAlign={alignText}>
+              <FormWrapper onSubmit={(data) => console.log(data)} cta={cta}>
+                <TextInput
+                  {...textinput}
+                  name="TextInput"
```

Comment:
Hardcoding label and placeholder because the typing for the `hook-fields` is incompatible with the payload types & needs to be updated